src/iuse.cpp

```diff
@@ -2339,21 +2169,30 @@ int iuse::radio_mod( player *p, item *, bool, const tripoint& )
 
 int iuse::remove_all_mods(player *p, item *, bool, const tripoint& )
 {
-    static const std::vector<std::string> removable_mods = {{
-        "DOUBLE_AMMO", "USE_UPS", "ATOMIC_AMMO"
-    }};
+    if( !p ) {
```

Comment:
True. I want to rework `iuse` to use an `item_location`. What would a good function signature look like?